### PR TITLE
Fix an issue with the generated KtxTargets.cmake 

### PR DESCRIPTION
**Pull request by UX3D-becher**
Fixes issue #326 

In this pull request we changed the problematic dependencies of the __ktx__ target in the root CMakeLists.txt from `PUBLIC` to `PRIVATE` and fixed configuration errors in other cmake files. These errors mainly stem from the use of the INTERFACE_INCLUDE_DIRECTORIES property of the __ktx__ that after the above change does not contain the private include paths any more.

### CODE REVIEW

**Comment by UX3D-nopper** [comment]
@MarkCallow Can u pls review?

**Comment by MarkCallow** [comment]
Please create an issue. Note that there is a large change coming to the CMake files to add a libktx_read that can only read KTX files. I Plan to merge that PR (#324) first. You will have to rebase this after the merge if, as I expect it to be, the bug is still present.

File: CMakeLists.txt
Comment (MarkCallow) [inline CMakeLists.txt:200]: I wonder if this line and all similar are really necessary. I've requested a review from @atteneder.

File: CMakeLists.txt
Comment (MarkCallow) [inline CMakeLists.txt:332]: This and related lines will have to change after PR #324 is merged as, if the built library is a static library(an option added by PR #324) libzstd will be required by the application.

File: CMakeLists.txt
Comment (MarkCallow) [inline CMakeLists.txt:406]: This is also going to be needed by the app when linking with a static library which is a problem as there is currently no separate release of the dfdutils library.

**Review by MarkCallow — CHANGES_REQUESTED** [review]

File: CMakeLists.txt
Comment (atteneder) [reply CMakeLists.txt:200]: While this changes are not necessary, I agree it makes sense to strictly differentiate between public and private(internal) headers.

File: CMakeLists.txt
Comment (atteneder) [reply CMakeLists.txt:332]: Wasn't aware of this. I'd need to test.

File: CMakeLists.txt
Comment (atteneder) [reply CMakeLists.txt:406]: Do users of KTX need to use dfdutils (dfd.h) directly commonly?

A clean solution could be to create a CMake config for dfdutils as well and link to its CMake target.
Users who need dfdutils directly could direclty refer to it.

File: interface/basisu_c_binding/CMakeLists.txt
Comment (atteneder) [inline interface/basisu_c_binding/CMakeLists.txt:25]: I'd appreciate consistent indentation. Same in tests.cmake:42

**Review by atteneder — APPROVED** [review]

File: CMakeLists.txt
Comment (MarkCallow) [reply CMakeLists.txt:332]: libktx has references to functions in libzstd. If you are building a shared library these will cause libzstd to be linked into the shared libktx. If static, the app needs to link with libzstd to satisfy the references.

File: CMakeLists.txt
Comment (MarkCallow) [reply CMakeLists.txt:200]: I am referring to the `INSTALL_INTERFACE` line. lib/basisu/transcoder is not installed. It is purely internal to the libktx. Hence my question.

There shouldn't be any need to actually make and install a transcoder library. If shared libktx is being used I don't think anybody cares if the whole libktx is loaded into memory. If a static library is being used and only the transcoder is referenced, the linker will link only the transcoder functions into the application.

File: CMakeLists.txt
Comment (MarkCallow) [reply CMakeLists.txt:406]: Quite possibly, if the app wishes to examine the DFD. At present only a small part of the information it contains is made available through the libktx API.

I agree we should make a CMake config for it, as you suggest @atteneder. dfdutils is included via `git subrepo` from https://github.com/KhronosGroup/dfdutils. I need to push some recent additions and fixes back there and we should add the CMake config there.

**Comment by MarkCallow** [comment]
PR #324 has been merged. Please rebase this against the new master, make the changes necessary for the static library case and remove lib/basisu/transcoder from the INSTALL_INTERFACE. We will work separately on a CMake config for the dfdutils. Note the dfdutils will be in the library archive sohat an application needs in order to use them are `dfd.h` and `khr_df.h`. We should add those to the installers and the INSTALL_INTERFACE here. Please add these changes to this PR.

**Comment by MarkCallow** [comment]
Please update this PR.

**Comment by MarkCallow** [comment]
Travis CI has not built this or the other open PR. I don't at present know why. There has been no change in the .travis.yml since the last time Travis did a PR build. I can't merge this until we have a successful build. If you have any ideas, please let me know.

**Comment by MarkCallow** [comment]
The missing Travis PR build has something to do with Khronos's Travis membership. It will probably take a few days to sort out. Please wait.

File: CMakeLists.txt
Comment (MarkCallow) [inline CMakeLists.txt:214]: Is this line really necessary? `lib/basisu/transcoder` is never installed.

File: CMakeLists.txt
Comment (MarkCallow) [inline CMakeLists.txt:366]: Is this line really necessary? `lib/dfdutil` is never installed.

File: CMakeLists.txt
Comment (MarkCallow) [inline CMakeLists.txt:441]: Is this line really necessary? `lib/basisu` is never installed.

**Review by MarkCallow — CHANGES_REQUESTED** [review]
I am repeating my question from my first review. Please answer.

**Comment by atteneder** [comment]
Sorry for late reply (I'm preparing a relocation myself now).
I try to have a look in the afternoon.

File: CMakeLists.txt
Comment (UX3D-becher) [reply CMakeLists.txt:214]: Yes, it is necessary. Otherwise the transcoder headers are defined as being part of the public interface of the ktx library target. As they are however not installed, this results in an error when cmake `find_package(Ktx)` validates the include directories of the imported target.

File: CMakeLists.txt
Comment (UX3D-becher) [reply CMakeLists.txt:366]: This is analogue to the transcoder headers

File: CMakeLists.txt
Comment (UX3D-becher) [reply CMakeLists.txt:441]: Same thing as with the transcoder headers, if they are not installed they must not be specified as being part of the interface.

**Review by MarkCallow — APPROVED** [review]

**Comment by MarkCallow** [comment]
Thanks for the explanation. Now we have to wait for Khronos to resolve the issue around Travis CI builds.

Note there is a PR #335 adding support for building on Android which makes many changes to the CMake files some of which I expect will overlap with these. I plan to merge this PR first as it has been open for some time. That means PR #335 will need rebasing.

**Comment by atteneder** [comment]
> Sorry for late reply (I'm preparing a relocation myself now).
> I try to have a look in the afternoon.

Oh, I see you already cleared everything up. Nonetheless I tested on Linux and found no issues!
Thanks!

**Comment by MarkCallow** [comment]
Khronos has resolved the issue with Travis CI. I triggered a redelivery of the webhook payload for the latest commit in this PR but it has not triggered a build. @UX3D-becher please push something to this PR to trigger a build so we can test the Linux, macOS and Web builds with these changes.

**Comment by MarkCallow** [comment]
Please do a push to trigger a Travis CI build so we can check the iOS, Linux & macOS builds. I want to get this PR merged.

**Comment by MarkCallow** [comment]
You need to make a small change and push it to this repo to trigger a Travis-CI build. I have tried to redelivering the webhook requests from the previous commits from GitHub but it is not working. Until we get successful iOS, Linux, macOS and web builds (all of which are done on Travis) I cannot merge this PR.

**Comment by MarkCallow** [comment]
@UX3D-becher trigger a build or else I will have to close this PR without merging.

**Comment by MarkCallow** [comment]
Been away a couple of days. Sorry for the delay. The change looks good to my limited knowledge of Android. Is zstd included as a standard feature of Android? I'm going to request a review from Andreas.

Unfortunately Khronos seems have used up all its credits with Travis CI so the builds have stopped again. It is infuriating. Trying to get it resolved. I may need you to push a commit again. Perhaps lack of credits is the reason re-sending the WebHook message for a previous commit failed to trigger a build. I'll try that again first once the problem is sorted.

**Review by MarkCallow — COMMENTED** [review]
Been away a couple of days. Sorry for the delay. The change looks good to my limited knowledge of Android. Is zstd included as a standard feature of Android? I'm going to request a review from Andreas.

**Comment by MarkCallow** [comment]
The problem blocking the macOS builds has reportedly been fixed, however redelivering GitHub webhook payloads has no visible effect so I have to again ask you to push another commit to this PR to trigger a build. Please do it as soon as possible.

**Comment by UX3D-becher** [comment]
I cleaned some whitespace to trigger a build, hopefully these changes were in your interest anyway

**Comment by MarkCallow** [comment]
Thanks for making it easier to use libktx is other projects.

**Pull request merged**
